substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/Target_java_lang_VirtualThread.java

```diff
@@ -165,6 +167,9 @@ void mount() {
         }
 
         carrier.setCurrentThread(asThread(this));
+        if (HasJfrSupport.get()) {
```

Comment:
Yes, that's right.  In the future maybe we should also check if the thread is excluded. I haven't added it in this PR because there are some related changes already in this one: https://github.com/oracle/graal/pull/6365